cmd/threatest/run.go

```diff
@@ -0,0 +1,221 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/datadog/threatest/pkg/threatest"
+	"github.com/datadog/threatest/pkg/threatest/parser"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"math"
+	"os"
+	"strconv"
+	"time"
+)
+
+// RunCommand implements the command to run Threatest test scenarios
+type RunCommand struct {
+	SSHConfig      *SSHConfiguration
+	InputFiles     []string
+	Parallelism    int
+	JsonOutputFile string
+}
+
+type SSHConfiguration struct {
+	SSHHost     string
+	SSHUsername string
+	SSHKey      string
+}
+
+type ScenarioRunResult struct {
+	Description     string    `json:"description"`
+	Success         bool      `json:"isSuccess"`
+	ErrorMessage    string    `json:"errorMessage"`
+	DurationSeconds float64   `json:"durationSeconds"`
+	TimeDetonated   time.Time `json:"timeDetonated"`
+	//TODO: We possibly want to add some metadata about the kind of detonation
+}
+
+func NewRunCommand() *cobra.Command {
+	var sshHost string
+	var sshUsername string
+	var sshKey string
+	var parallelism int
+	var jsonOutputFile string
+
+	runCmd := &cobra.Command{
+		Use:          "run",
+		Short:        "Run Threatest scenarios",
+		SilenceUsage: true,
+		Example:      "run /path/to/scenario/1 [/path/to/scenario/2]...",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			command := RunCommand{
+				InputFiles:     args,
+				Parallelism:    parallelism,
+				JsonOutputFile: jsonOutputFile,
+				SSHConfig: &SSHConfiguration{
+					SSHHost:     sshHost,
+					SSHUsername: sshUsername,
+					SSHKey:      sshKey,
+				},
+			}
+
+			return command.Do()
+		},
+	}
+
+	runCmd.Flags().StringVarP(&sshHost, "ssh-host", "", os.Getenv("THREATEST_SSH_HOST"), "SSH host to connect to for remote command detonation. Can also be specified through THREATEST_SSH_HOST")
+	runCmd.Flags().StringVarP(&sshUsername, "ssh-username", "", os.Getenv("THREATEST_SSH_USERNAME"), "SSH username to use for remote command detonation  (leave empty to use system configuration). Can also be specified through THREATEST_SSH_USERNAME")
+	runCmd.Flags().StringVarP(&sshKey, "ssh-key", "", os.Getenv("THREATEST_SSH_KEY"), "SSH keypair to use for remote command detonation (leave empty to use system configuration). Can also be specified through THREATEST_SSH_KEY. Only unencrypted keys are currently supported")
+	runCmd.Flags().StringVarP(&jsonOutputFile, "output", "o", "", "Write JSON test results to the specified file")
+
+	var defaultParallelism = 5
+	if parallelism, isSet := os.LookupEnv("THREATEST_MAX_PARALLELISM"); isSet {
+		parsedParallelism, err := strconv.Atoi(parallelism)
+		if err != nil {
+			log.Fatalf("unable to convert max parallelism '%s' to integer: %v", parallelism, err)
+		}
+		defaultParallelism = parsedParallelism
+	}
```

Comment:
[nit] maybe worth putting this logic into a separate function to keep the `NewRunCommand` body as homogeneous and simple as possible?

Reply:
addressed in bac096c935645bdf1a72a4120786fc6df954788a